src/components/Autosuggest/Autosuggest.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import Autosuggest from "react-autosuggest";
+import Label from "../Label/Label";
+import Input from "../Input/Input";
+import { InputTypes } from "../Input/InputTypes";
+import HelperErrorText from "../HelperErrorText/HelperErrorText";
+
+const AutosuggestComp = ({
+    defaultValue,
+    childRef,
+    list,
+    fieldName,
+    renderSuggestion,
+    getSuggestionValue,
+    getSuggestions,
+    placeholder,
+    label,
+    helperText,
+    isRequired = false,
+}: any) => {
+    const [value, setValue] = useState(defaultValue);
+    const [suggestions, setSuggestions] = useState([]);
+
+    const onChange = (event, { newValue }) => setValue(newValue);
+
+    // Autosuggest will call this function every time suggestions need to be
+    // updated. `getSuggestions` must be passed.
+    const onSuggestionsFetchRequested = ({ value }) =>
+        setSuggestions(getSuggestions(value, list));
+
+    const onSuggestionsClearRequested = () => setSuggestions([]);
+
+    // Autosuggest will pass through all these props to the Input component.
```

Comment:
This looks like it might go against the composeability principles - we really want a compound component to allow the user to pass in their own `<Label>`, `<Input>` and `<HelperErrorText>` rather than having a component that's doing them for you (for example, what happens if the user doesn't want a label?)  

To this end, I think this should be a standalone `<Input-autosuggest>` component that doesn't reference `<Input>` (or they both reference the same helpers)

Reply:
There's a part of me that really wants 
`<Input autosuggest={true}         
       getSuggestionValue={blah}
        getSuggestions={blah2}
        list={listOfThings}  />` but it might lead us back to the "too many props" problem we had before